ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/KsqlResource.java

```diff
@@ -300,23 +320,30 @@ private SourceDescription describe(String name, boolean extended) throws KsqlExc
 
     StructuredDataSource dataSource = ksqlEngine.getMetaStore().getSource(name);
     if (dataSource == null) {
-      throw new KsqlException(String.format("Could not find STREAM/TABLE '%s' in the Metastore", name));
+
+      PersistentQueryMetadata metadata = ksqlEngine.getPersistentQueries().get(new QueryId(name));
+      // check if we are describing a Query (not found in the metastore)
+      if (metadata == null) {
+        throw new KsqlException(String.format("Could not find STREAM/TABLE/QUERY '%s' in the Metastore", name));
+      }
+      KsqlStructuredDataOutputNode outputNode = (KsqlStructuredDataOutputNode) metadata.getOutputNode();
+      return new SourceDescription(outputNode, metadata.getStatementString(), name, metadata.getTopoplogy(), metadata.getExecutionPlan(), true, ksqlEngine.getTopicClient());
     }
 
     List<PersistentQueryMetadata> queries = ksqlEngine.getPersistentQueries().values().stream().filter(meta -> ((KsqlStructuredDataOutputNode) meta.getOutputNode()).getKafkaTopicName().equals(dataSource.getKsqlTopic().getTopicName())).collect(Collectors.toList());
```

Comment:
This is from your previous patch, but there are lots of very very long lines. Makes it hard to review and hard to read in the code later. Can we update the code to stick to the 120 character line length?

Reply:
yep